CIP-0129/README.md

```diff
@@ -0,0 +1,161 @@
+
+We define a byte format to combine the tx ID and the index to form a single valid byte string, as such it can be converted into a hex format and have its own bech32 prefix.
+
+Transaction ID is always a 32 byte length, hence we can append the index bytes to the tx id, please see below
+
+**Example 1**
+TX ID in Hex - `0000000000000000000000000000000000000000000000000000000000000000`
+Gov Action index in HEX - `11` (number 17)
+gov action ID - `000000000000000000000000000000000000000000000000000000000000000011`
+bech32 - `gov_action1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqpzklpgpf`
+
+**Example 2**
+TX ID in Hex - `1111111111111111111111111111111111111111111111111111111111111111`
+Gov Action index in HEX - `00` (number 0) 
+gov action ID - `111111111111111111111111111111111111111111111111111111111111111100`
+bech32 - `gov_action1zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zygsq6dmejn`
+
+### Bech32 Encoding
+
+| Prefix        | Meaning                                                 | Contents                                                          |
+| ------------- | --------------------------------------------------------| ----------------------------------------------------------------- |
+| `drep`        | DRep identifier                                         | DRep Credential                                                   |
+| `cc_hot`      | CC Hot identifier                                       | CC Hot Credential                                                 |
+| `cc_cold`     | CC Cold identifier                                      | CC Cold Credential                                                |
+| `gov_action`  | gov action identifier                                   | gov action tx id concatenated with index                          |
+
+
+### Identifier Test Vector
+
+We can define a complete identifier as per the spec above by combining the header and the key, see below
+
+#### Constitutional Committee Hot Credential
+
+Key - `00000000000000000000000000000000000000000000000000000000`
+Type - `Key Hash`
+
+Identifier - `0200000000000000000000000000000000000000000000000000000000`
+
+Bech32 - `cc_hot1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqvcdjk7`
+
+#### Constitutional Committee Cold Credential
+
+Key - `00000000000000000000000000000000000000000000000000000000`
+Type - `Script Hash`
+
+Identifier - `1300000000000000000000000000000000000000000000000000000000`
+
+Bech32 - `cc_cold1zvqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6kflvs`
+
+#### DRep Credential
+
+Key - `00000000000000000000000000000000000000000000000000000000`
+Type - `Key Hash`
+
+Identifier - `2200000000000000000000000000000000000000000000000000000000`
+
+Bech32 - `drep1ygqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq7vlc9n`
+
+## Rationale: how does this CIP achieve its goals?
```

Comment:
I believe it would be valuable to describe the ways that this differs from the existing CIPs, and why this approach is preferable.

Reply:
@Ryun1 detailed in the Rational and Motivation

Reply:
I found the Abstract and Motivation might have been missing some background in order (for me) to to understand the issue here.  Let me try to summarize in order to verify my understanding:

1.  CIP-105 defines bech32 identifiers with the key (CC hot, CC cold, DREP) and credential (script hash, public key hash) defined by the bech32 prefix only.  In CIP-105 the binayr payload for the bech32 encoding is always just the key material itself.

2.  I am inferring from the discussion thread that various infrastructure and tools that use these identifiers decode the bech32 identifier and store the key material internally as the bytes themselves (presumably for space and efficiency reaons), thereby losing the metadata that was encoded in the bech32 prefix specified in CIP-105.

3.  This CIP-129 defines a single byte header to be prefixed to the key material prior to bech32 encoding that would include this metadata.  Note this would also be slightly redundant if the CIP-105 prefixes are used.

Since the primary purpose of bech32 encoding is for human-readable transmission, it does make sense to add a binary level encoding of the key type and credential metadata.  Unless I'm missing something it also seems that the single byte header can be inferred from CIP-105 -> CIP-129 if it is missing, through the bech32 prefix.